### PR TITLE
AZP: add fedora 34

### PR DESCRIPTION
**Pull request by avildema**
add fedora 34 container for build testing

### CODE REVIEW

**Review by yosefe — APPROVED** [review]

File: buildlib/tools/builds.sh [hunk 1]
```diff
@@ -8,6 +8,10 @@ source ${realdir}/../az-helpers.sh
 # Build documentation
 #
 build_docs() {
+	if [ `cat /etc/system-release | grep -i "fedora release 34" |wc -l` -gt 0 ]; then
```
Comment (yosefe) [inline buildlib/tools/builds.sh:11]: minor: space after the last `|`

File: buildlib/tools/builds.sh [hunk 2]
```diff
@@ -8,6 +8,10 @@ source ${realdir}/../az-helpers.sh
...
+		echo "==== Skip build docs on Fedora 34 ===="
```
Comment (yosefe) [inline buildlib/tools/builds.sh:12]: can we use az_log_warning ?

File: docs/doxygen/ucxdox [hunk 3]
```diff
@@ -234,12 +234,6 @@ TAB_SIZE               = 4
 
 ALIASES                =
 
-# This tag can be used to specify a number of word-keyword mappings (TCL only).
-# A mapping has the form "name=value". For example adding "class=itcl::class"
-# will allow you to use the command class in the itcl::class meaning.
-
-TCL_SUBST              =
-
```
Comment (yosefe) [inline docs/doxygen/ucxdox:242]: still need to remove these?

**Comment by avildema** [comment]
> @avildema the test passes now

I have updated Fedora image

File: buildlib/tools/builds.sh [hunk 4]
```diff
@@ -8,8 +8,8 @@ source ${realdir}/../az-helpers.sh
 # Build documentation
 #
 build_docs() {
-	if [ `cat /etc/system-release | grep -i "fedora release 34" |wc -l` -gt 0 ]; then
-		echo "==== Skip build docs on Fedora 34 ===="
+	if [ `cat /etc/system-release | grep -i "fedora release 34" | wc -l` -gt 0 ]; then
+		azure_log_warning "==== Skip build docs on Fedora 34 ===="
```
Comment (yosefe) [inline buildlib/tools/builds.sh:12]: pls remove "====" stuff

**Review by yosefe — APPROVED** [review]

**Pull request merged**